web/src/stores/workflow-store.ts

```diff
@@ -372,6 +381,36 @@ export const useWorkflowStore = create<WorkflowStoreState>((set, get) => ({
 			}),
 		);
 	},
+	hasMissingSecret: (nodeId: string, secretIdx?: number) => {
+		const missingSecrets = get().missingSecretForNodes;
+		if (missingSecrets) {
+			if (secretIdx !== undefined) {
+				return (
+					missingSecrets.has(nodeId) &&
+					missingSecrets.get(nodeId)!.has(secretIdx)
+				);
+			}
+			return missingSecrets.has(nodeId);
+		}
+		return false;
```

Comment:
Simplification suggestion:

```suggestion
		return !!(secretIdx === undefined ? missingSecrets?.has(nodeId) : (missingSecrets?.get(nodeId)?.has(secretIdx));
```

Note: `secretIdx` should be replaced with `secretMappingPlaceholder` (see comment above)